doc/ansible/dev/apb_migration_guide.md

```diff
@@ -0,0 +1,241 @@
+
+The `version` and `group`, will remain unchanged, but update the `kind` with a `Binding` suffix. 
+
+For example, if you have a resource with `kind: Memcached`, the kind of your new entry will be `MemcachedBinding`. 
+
+The playbook for this entry should map to your `bind` playbook, (likely location `/opt/ansible/playbooks/bind.yml`), and if you have an `unbind` playbook then set the playbook for the finalizer to point to it (likely location `/opt/ansible/playbooks/unbind.yml`). If you don't have an `unbind` playbook, remove the finalizer block for your `Binding` resource.
+
+For an APB with both `bind` and `unbind` playbooks, the `watches.yaml` would end up looking like this:
+
+```yaml
+---
+- version: v1alpha1
+  group: apps.example.com
+  kind: Memcached
+  playbook: /opt/ansible/playbooks/provision.yml
+  finalizer:
+    name: finalizer.memcached.apps.example.com/v1alpha1
+    playbook: /opt/ansible/playbooks/deprovision.yml
+- version: v1alpha1
+  group: apps.example.com
+  kind: MemcachedBinding
+  playbook: /opt/ansible/playbooks/bind.yml
+  finalizer:
+    name: finalizer.memcachedbinding.apps.example.com/v1alpha1
+    playbook: /opt/ansible/playbooks/unbind.yml
+```
+
+
+You will also need to run `operator-sdk add crd --api-version=<group>/<version> --kind=<kind>` to generate a new CRD and example in `deploy/crds`.
+
+#### deploy/crds/
+Now that you have all your CRDs created, you can generate the OpenAPI spec for them using your `apb.yml`. 
+
+The `convert.py` script included at the bottom of this document can handle the conversion to the OpenAPI spec, at which point you can copy paste everything from `validation:` on into your primary CRD (for the regular `parameters`), or into your `Binding` CRD (for `bind_parameters`).
+
+You may notice that the OpenAPI validation uses `camelCase` parameters, while your `apb.yml` and Ansible playbooks probably assume `snake_case` variables. `Ansible Operator` will automatically convert the `camelCase` parameters from the Kubernetes resource into `snake_case` before passing them to your playbook, so this should not require any change on your part.
+
+### Ansible logic
+There will be some changes required to your Ansible playbooks/roles/tasks.
+
+#### Idempotence
+
+One major conceptual difference between the APB model and the Operator model, is that APBs are meant to run `provision` once, while operators constantly reconcile to ensure that the state of the cluster matches the state that the user requested. 
+
+This means that you will need to ensure that your playbooks are idempotent, and can be run repeatedly with the same parameters without causing an error.
+
+#### Service Bundle contract and meta variables
+Ansible Operator does not respect the Service Bundle contract that exists between APBs and the Ansible Service Broker. The following variables will not be passed in by the Ansible Operator:
```

Comment:
Do we need to explain why the contract is not supported? Not quite sure what to say, just feels like the Ansible Operator doesn't support them because we didn't want to support them. In reality, it's a completely different paradigm. When I read it as written, I come away with the notion that the Ansible Operator developers just wanted not to support the contract which is definitely not true.

If we don't have a better way to say that, that's okay, since I don't really have a suggestion.